selfdrive/modeld/models/driving.cc

```diff
@@ -46,8 +46,29 @@ constexpr int OUTPUT_SIZE =  POSE_IDX + POSE_SIZE;
   constexpr int TEMPORAL_SIZE = 0;
 #endif
 
+// Keep a running history of meta states for the disengage alert
+float engaged_history[250];
+float steering_history[250];
+float rolling_engaged_prob = 0;
+float rolling_steering_prob = 0;
+float engaged_threshold_mse = 0;
+float steering_threshold_mse = 0;
+uint64_t last_desire_frame = 0;
+uint64_t last_blinker_frame = 0;
+uint64_t last_steering_frame = 0;
+uint64_t history_size = 0;
+
 // #define DUMP_YUV
 
+float smooth(float p, float a, float b) {
+  return p * a + (1 - p) * b;
+}
+
+float threshold_mse(float x, float threshold, bool gt) {
```

Comment:
What does gt mean?

Reply:
greater than / less than, it's a flag to control whether the error should be taken above or below the threshold. I can split this up into two separate functions to make it more readable, or maybe just inline it.

Reply:
Yeah two functions sounds more readable